---
comment_id: 1721585058012
is_spam: false
is_deleted: false
post: /en/clojure-with-retry/
date: 2024-07-21 18:04:18 +0000
author_fullname: 'The Alchemist'
---

I've seen this a lot in enterprise code too, in other programming languages.

Errors fall into different categories, as you say.  Exceptions are usually, but always, the error-reporting mechanism, even if something like Integer.parseInt.  Exceptions get wrapped, and often the "true" cause of an exception is unknown unless you have some intelligent .getCause() analysis.

Some errors need clean up (corrupt file created), some should be retried (throttle the request), some should be ignored-but-logged (unknown user tried accessing a document).

On some exceptions, you should just bail and re-start the process (OutofMemoryError).

Unless you have a firm grasp of the transitive errors a function can cause, it's hard to perform retries.
